DiscordBot.DataAccess/EventsSheetsService.cs

```diff
@@ -254,11 +258,104 @@ public async Task<IEnumerable<DiscordEvent>> ListEventsAsync()
             }
         }
 
-#pragma warning disable 1998 // Disable compiler warning stating that the unimplemented functions are synchronous
+        public async Task<IEnumerable<EventResponse>> GetEventResponseOptionsAsync(int eventKey)
+        {
+            try
+            {
+                var request = sheetsService.Spreadsheets.Values.Get(
+                    spreadsheetId,
+                    $"{eventKey}!1:2"
+                );
+                request.ValueRenderOption = GetRequest.ValueRenderOptionEnum.FORMATTEDVALUE;
+
+                var sheetsResponse = await request.ExecuteAsync();
+
+                if (sheetsResponse == null || sheetsResponse.Values.Count < 2)
+                {
+                    throw new EventsSheetsInitialisationException($"Event sheet {eventKey} is empty");
+                }
+
+                return sheetsResponse.Values[0]
+                    .Zip(sheetsResponse.Values[1])
+                    .Skip(1) // Skip titles column
+                    .Select<(object emoji, object name), EventResponse>(response =>
+                        new EventResponse((string)response.emoji, (string)response.name)
+                    );
+            }
+            catch (GoogleApiException)
+            {
+                throw new EventInitialisationException(
+                    $"Could not find event responses for event {eventKey}. Has it been published yet?"
+                );
+            }
+        }
+
         public async Task AddResponseForUserAsync(int eventKey, ulong userId, string responseEmoji)
         {
+            var responseRowTask = GetResponseRowNumber(eventKey, userId);
+            var responseColumnsTask = GetEventResponseOptionsAsync(eventKey);
+
+            await Task.WhenAll(responseRowTask, responseColumnsTask);
+            var responseRow = await responseRowTask;
+            var responseColumns = await responseColumnsTask;
```

Comment:
Do you need to await these again given that you've just awaited them both in your `WhenAll` statement above? 🙂

Reply:
I need to unwrap them in some way. I could use `*Task.Result` instead maybe? What do you think? 🙂

Reply:
Using `.Result` sounds sensible to me 🙂

Reply:
Changed in c520c27 🙂